middleware.ml

```diff
@@ -30,55 +68,60 @@ let redirect_to_dashboard reqd ?(msg = "") () =
   Httpaf.Reqd.respond_with_string reqd response msg;
   Lwt.return_unit
 
-let auth_middleware ~users handler reqd =
-  let headers = (Httpaf.Reqd.request reqd).headers in
-  match Httpaf.Headers.get headers "Cookie" with
-  | Some cookies -> (
-      let cookie_list = String.split_on_char ';' cookies in
-      let session_cookie =
-        List.find_opt
-          (fun cookie ->
-            let parts = String.trim cookie |> String.split_on_char '=' in
-            match parts with
-            | [ name; _ ] -> String.equal name "molly_session"
-            | _ -> false)
-          cookie_list
-      in
-      match session_cookie with
-      | Some cookie -> (
-          let parts = String.trim cookie |> String.split_on_char '=' in
-          let value = List.nth parts 1 in
-          match User_model.find_user_by_key value users with
-          | Some user -> (
-              let user_session =
-                List.find_opt
-                  (fun (cookie : User_model.cookie) ->
-                    String.equal cookie.name "molly_session")
-                  user.cookies
-              in
-              match user_session with
-              | Some cookie -> (
-                  match String.equal cookie.value value with
-                  | true -> handler reqd
-                  | false ->
-                      Logs.err (fun m ->
-                          m
-                            "auth-middleware: Session value doesn't match user \
-                             session %s\n"
-                            value);
-                      redirect_to_login reqd ())
-              | None ->
+let cookie_value_from_auth_cookie ~cookie =
+  let parts = String.trim cookie |> String.split_on_char '=' in
+  List.nth parts 1
```

Comment:
can we have a `match String.split_on_char ..` and some error handling (logging?)